### DISCUSSION TITLE
Row Filters

### DISCUSSION THREAD

**Discussion by Niyanthan** [fbaligand/kibana-enhanced-table#218] [Q&A] [answered]
I am new to Kibana and enhanced table.
I have a calculated field with values 0 or 1 or 2. I want to filter out those rows with 0 and dhow only rows with value 1 or 2?
I have read about row filtering but  unable to arrive a solution to use this as per my requirement mentioned above.

Thanks in advance for helping me out!

**Comment by fbaligand** [comment]
You just have to fill « row computed filter » setting with : 
col0 != 0

given that your column is the first column.

**Comment by Niyanthan** [comment]
Does Row computed filter works for computed column?
Because mine is a computed column and when I set row computed filter with col36 != 0 in my case it is giving me this error "Column number 36 does not exist, in Rows computed filter: col36 != 0”

**Comment by fbaligand** [comment, marked as answer]
Waou... 36 columns! That's a lot!
I'm curious: you use "Enhanced Table" visualization (for aggregated data) or "Document Table" visualization (for single documents)

To answer your question: Yes, "rows computed filter" works nice with computed columns. I just verified to be sure.
But be sure that you use 0-based index. I mean: if you want the 36th column, so use `col35` (not `col36`).
By the way, you can also reference the column by its label: `col['Column Label']`

**Comment by Niyanthan** [comment]
Let me try once again yes I used 0-based index. Am using enhanced table for User Account prioritisation which is calculated upon multiple categories. Even though have numerous column I could achieve the result using enhanced table very easily. Thanks a lot to you for giving us this plugin.

**Comment by Niyanthan** [comment]
Also in Cell computed CSS want to differentiate the colors based on cell value, For example if cell value is 2 I want the cell to be green color and if its value is 1 I want it to be red color so I tried the the following statement

value=2?”background-color:green”:”background-color:red”

My output was green for all the cells in that column.

**Reply by fbaligand** [reply]
You have to use `==` operator, not `=` operator.

So, in your case:
`(value == 2) ? "background-color:green" : "background-color:red"`

**Reply by Niyanthan** [reply]
Thank you it works fine
